include/onnxruntime/core/session/onnxruntime_cxx_api.h

```diff
@@ -7,16 +7,23 @@
 #include <string>
 #include <stdexcept>
 #include <memory>
+#include "core/common/exceptions.h"
 
 //TODO: encode error code in the message?
-#define ORT_THROW_ON_ERROR(expr)                                       \
-  do {                                                                 \
-    OrtStatus* onnx_status = (expr);                                   \
-    if (onnx_status != nullptr) {                                      \
-      std::string ort_error_message = OrtGetErrorMessage(onnx_status); \
-      OrtReleaseStatus(onnx_status);                                   \
-      throw std::runtime_error(ort_error_message);                     \
-    }                                                                  \
+#define ORT_THROW_ON_ERROR(expr)                                                   \
+  do {                                                                             \
+    OrtStatus* onnx_status = (expr);                                               \
+    if (onnx_status != nullptr) {                                                  \
+      std::string ort_error_message = OrtGetErrorMessage(onnx_status);             \
+      OrtErrorCode error_code = OrtGetErrorCode(onnx_status);                      \
+      OrtReleaseStatus(onnx_status);                                               \
+      switch (error_code) {                                                        \
+        case ORT_NOT_IMPLEMENTED:                                                  \
+          throw ::onnxruntime::NotImplementedException(ort_error_message);         \
```

Comment:
>:: [](start = 16, length = 2)

nit: Using the '::' prefix shouldn't be necessary. We should not have any 'onnxruntime' namespaces nested within another, so just 'onnxruntime::' should be unambiguous.

Reply:
This is a public API file.  We won't do that, but the other people would.